vega_sim/scenario/common/agents.py

```diff
@@ -1581,22 +1582,30 @@ def initialise(
         self.vega.wait_for_total_catchup()
 
     def step(self, vega_state: VegaState):
+        market_info = self.vega.market_info(market_id=self.market_id)
+        market_in_auction = (
+            not market_info.trading_mode
+            == markets_protos.Market.TradingMode.TRADING_MODE_CONTINUOUS
+        )
         position = self.vega.positions_by_market(
             wallet_name=self.wallet_name, market_id=self.market_id
         )
         current_position = int(position[0].open_volume) if position else 0
         trade_side = (
             vega_protos.SIDE_BUY if current_position < 0 else vega_protos.SIDE_SELL
         )
-        if current_position:
-            self.vega.submit_market_order(
-                trading_wallet=self.wallet_name,
-                market_id=self.market_id,
-                side=trade_side,
-                volume=np.abs(current_position),
-                wait=True,
-                fill_or_kill=False,
-            )
+        if (not market_in_auction) and current_position:
+            try:
+                self.vega.submit_market_order(
+                    trading_wallet=self.wallet_name,
+                    market_id=self.market_id,
+                    side=trade_side,
+                    volume=np.abs(current_position),
+                    wait=True,
+                    fill_or_kill=False,
+                )
+            except OrderRejectedError:
+                print("Order rejected.")
```

Comment:
We should use a logger here

Reply:
Yeah I think so, it's an error which you may or may not care about so doing at exception level might well feel noisy. Could do at `warn` but I never much like `warn`